src/main/java/net/dries007/tfc/common/blocks/LargeVesselBlock.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Override
+    public boolean canSurvive(BlockState state, LevelReader level, BlockPos pos)
+    {
+        BlockPos belowPos = pos.below();
+        BlockState belowState = level.getBlockState(belowPos);
+        return !belowState.isAir() && belowState.isFaceSturdy(level, belowPos, Direction.UP) && super.canSurvive(state, level, pos);
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext context)
+    {
+        return context.getItemInHand().getTag() != null ? defaultBlockState().setValue(SEALED, true) : defaultBlockState();
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public InteractionResult use(BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit)
+    {
+        if (player.isShiftKeyDown())
+        {
+            toggleSeal(level, pos, state);
+        }
+        else
+        {
+            level.getBlockEntity(pos, TFCBlockEntities.LARGE_VESSEL.get()).ifPresent(vessel -> {
+                if (player instanceof ServerPlayer serverPlayer)
+                {
+                    NetworkHooks.openGui(serverPlayer, vessel, pos);
+                }
+            });
+        }
+        return InteractionResult.SUCCESS;
+    }
+
+    @Override
+    public ItemStack getCloneItemStack(BlockState state, HitResult target, BlockGetter level, BlockPos pos, Player player)
+    {
+        final ItemStack stack = super.getCloneItemStack(state, target, level, pos, player);
+        if (state.getValue(SEALED))
+        {
+            final BlockEntity entity = level.getBlockEntity(pos);
+            if (entity instanceof InventoryBlockEntity<?> inv)
+            {
+                inv.saveToItem(stack);
+            }
+        }
+        return stack;
+    }
+
+    @Override
+    public Size getSize(ItemStack stack)
+    {
+        return stack.getTag() == null ? Size.VERY_LARGE : Size.HUGE;
+    }
+
+    @Override
+    public Weight getWeight(ItemStack stack)
+    {
+        return Weight.VERY_HEAVY;
+    }
+
+    @Override
+    protected void beforeRemove(InventoryBlockEntity<?> entity)
+    {
+        if (!entity.getBlockState().getValue(SEALED))
+        {
+            entity.ejectInventory();
+        }
+        entity.invalidateCapabilities();
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable BlockGetter level, List<Component> tooltip, TooltipFlag flag)
+    {
+        final CompoundTag tag = stack.getTagElement(Helpers.BLOCK_ENTITY_TAG);
+        if (tag != null)
+        {
+            // Decode the contents of the vessel
+            final ItemStackHandler inventory = new ItemStackHandler();
+
+            inventory.deserializeNBT(tag.getCompound("inventory"));
+
+            tooltip.add(new TranslatableComponent("tfc.tooltip.contents").withStyle(ChatFormatting.DARK_GREEN));
+            Helpers.addInventoryTooltipInfo(inventory, tooltip);
```

Comment:
Add a `if (Helpers.isEmpty(inventory))` to the tooltips here, so it doesn't show "Contents" when there's no content. i.e. [small vessels](https://github.com/TerraFirmaCraft/TerraFirmaCraft/blob/25de245d360c56502a9c4758dff059384d39dc90/src/main/java/net/dries007/tfc/common/items/VesselItem.java#L188).